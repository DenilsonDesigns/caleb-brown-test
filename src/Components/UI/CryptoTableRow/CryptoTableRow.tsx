import { useHistory } from "react-router";
import { convertVal } from "../../../util/globalFuncs";
import { MAIN_VIEW_ROW_KV_PAIRS } from "../../../util/globalVars";

const CryptoTableRow = (props: CryptoTableRowProps): JSX.Element => {
  const history = useHistory();

  console.log(props.coin);

  return (
    <tr
      onClick={() => {
        history.push(`/detail/${props.coin.id}`);
      }}
    >
      <th>{props.coin.market_cap_rank}</th>
      {Object.values(MAIN_VIEW_ROW_KV_PAIRS).map((value: string, i: number) => {
        return (
          <td key={i}>{convertVal(props.coin[value], value, props.coin)}</td>
        );
      })}
    </tr>
  );
};

export default CryptoTableRow;
