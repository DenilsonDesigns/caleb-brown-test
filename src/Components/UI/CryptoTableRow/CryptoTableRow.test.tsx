import { render, screen } from "@testing-library/react";
import CryptoTableRow from "./CryptoTableRow";
import { BrowserRouter } from "react-router-dom";

const COIN_PROP = {
  id: "dogecoin",
  symbol: "doge",
  name: "Dogecoin",
  circulating_supply: 131433118833.266,
  current_price: 0.208089,
  market_cap: 27284951939,
  market_cap_rank: 10,
  total_volume: 925001850,
  price_change_24h: -0.003380898453,
  price_change_percentage_24h: -1.59876,
  max_supply: null,
  total_supply: null,
};

let cryptoTableRow: any;

describe("CryptoTableRow:", () => {
  beforeEach(() => {
    cryptoTableRow = render(
      <BrowserRouter>
        <table>
          <tbody>
            <CryptoTableRow coin={COIN_PROP} />
          </tbody>
        </table>
      </BrowserRouter>
    );
  });

  it("matches snapshot", () => {
    expect(cryptoTableRow).toMatchSnapshot();
  });

  it("correctly renders info", () => {
    let coinTitle = screen.getByText(/DogeCoin/i);

    expect(coinTitle).toBeInTheDocument();

    let marketCap = screen.getByText(/27,284,951,939/i);
    expect(marketCap).toBeInTheDocument();
  });
});
