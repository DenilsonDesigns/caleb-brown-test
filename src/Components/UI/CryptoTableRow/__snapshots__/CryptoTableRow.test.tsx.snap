// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoTableRow: matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr>
            <th>
              10
            </th>
            <td>
              Dogecoin
            </td>
            <td>
              $0.21
            </td>
            <td>
              -1.59876
            </td>
            <td>
              $27,284,951,939.00
            </td>
            <td>
              131,433,118,833.3
            </td>
            <td>
              N/A
            </td>
            <td>
              N/A
            </td>
            <td>
              925,001,850
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr>
          <th>
            10
          </th>
          <td>
            Dogecoin
          </td>
          <td>
            $0.21
          </td>
          <td>
            -1.59876
          </td>
          <td>
            $27,284,951,939.00
          </td>
          <td>
            131,433,118,833.3
          </td>
          <td>
            N/A
          </td>
          <td>
            N/A
          </td>
          <td>
            925,001,850
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
